File: src/main/java/imaxct/domain/StudentCoursePK.java
package imaxct.domain;

import javax.persistence.Embeddable;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.io.Serializable;

/**
 * Created by maxct on 2017/4/9.
 */
@Embeddable
public class StudentCoursePK implements Serializable{
    public StudentCoursePK(){}

    public StudentCoursePK(Student sid, Course cid){
        this.sid = sid;
        this.cid = cid;
    }
    @ManyToOne
    @JoinColumn
    private Student sid;

    @ManyToOne
    @JoinColumn
    private Course cid;

    public Student getSid() {
        return sid;
    }

    public void setSid(Student sid) {
        this.sid = sid;
    }

    public Course getCid() {
        return cid;
    }

    public void setCid(Course cid) {
        this.cid = cid;
    }
}

File: src/main/java/imaxct/domain/StudentCoursePK.kt
package imaxct.domain

import javax.persistence.Embeddable
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import java.io.Serializable

/**
 * Created by maxct on 2017/4/9.
 */
@Embeddable
class StudentCoursePK : Serializable {
    constructor() {}

    constructor(sid: Student, cid: Course) {
        this.sid = sid
        this.cid = cid
    }

    @ManyToOne
    @JoinColumn
    var sid: Student? = null

    @ManyToOne
    @JoinColumn
    var cid: Course? = null
}
